packages/gatsby/src/db/common/query.ts

```diff
@@ -0,0 +1,206 @@
+import * as _ from "lodash"
+
+export interface DbQueryQuery {
+  type: "query"
+  path: Array<string>
+  query: DbFilterStatement
+}
+
+export interface DbQueryElemMatch {
+  type: "elemMatch"
+  path: Array<string>
+  nestedQuery: DbQuery
+}
+
+export type DbQuery = DbQueryQuery | DbQueryElemMatch
+
+export enum DbComparator {
```

Comment:
Do these need the `$` prefix? Seems redundant. (Unless it's for greppability or something, that would make sense to me)

Reply:
I want it to match the way it should be for sift.

Reply:
:man_shrugging: ok :)